core/signeddata.go

```diff
@@ -320,6 +322,12 @@ func (p VersionedSignedProposal) MessageRoot() ([32]byte, error) {
 			return p.DenebBlinded.Message.HashTreeRoot()
 		}
 
+		if p.IsGnosis {
```

Comment:
Wondering if start dealing with blinded blocks, shouldn't this `IsGnosis` come first, before `.Blinded` check?

Reply:
What if instead of this flag we simply test for `featureset.Enabled(featureset.GnosisBlockHotfix)`?

Reply:
Implemented!